packages/matter.js/src/cluster/SwitchCluster.ts

```diff
@@ -0,0 +1,153 @@
+/**
+ * @license
+ * Copyright 2022-2023 Project CHIP Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Attribute, Cluster, ClusterExtend, OptionalEvent, EventPriority } from "./Cluster.js";
+import { TlvUInt8 } from "../tlv/TlvNumber.js";
+import { MatterApplicationClusterSpecificationV1_0 } from "../spec/Specifications.js";
+import { BitFlag } from "../schema/BitmapSchema.js";
+import { TlvField } from "../tlv/TlvObject.js";
+
+/**
+ * Exposes interactions with a switch device for the purpose of using those interactions by other devices
+ *
+ * @see {@link MatterApplicationClusterSpecificationV1_0} § 1.11
+ */
+export const LatchingSwitchCluster = Cluster({
+    id: 0x3b,
+    name: "Switch",
+    revision: 1,
+
+    /** @see {@link MatterApplicationClusterSpecificationV1_0} § 1.11.4 */
+    features: {
+        /**  Indicates a switch that maintains its position after being pressed (or turned) */
+        latchingSwitch: BitFlag(0), // TODO: Validate when set, momentarySwitch is NOT set.
+
+        /** Indicates a switch that does not maintain its position after being pressed (or turned)  */
+        momentarySwitch: BitFlag(1), // TODO: Validate when set, latchingSwitch is NOT set.
+
+        /** Indicates a momentary switch that can distinguish and report release events. */
+        momentarySwitchRelease: BitFlag(2), // TODO: Validate when set, momentarySwitch is set.
+
+        /** Indicates a momentary switch that can distinguish and report long presses from short presses. */
+        momentarySwitchLongPress: BitFlag(3), // TODO: Validate when set, momentarySwitch and momentarySwitchRelease are both set.
+
+        /** 
+         * Idicates a momentary switch that can distinguish and report double press 
+         *  and potentially multiple presses with more events, such as triple press, etc. 
+         */
+        momentarySwitchMultiPress: BitFlag(4), // TODO: Validate when set, momentarySwitch and momentarySwitchRelease are both set.
+    },
+    supportedFeatures: {
+        latchingSwitch: true,
+        momentarySwitch: false,
+        momentarySwitchRelease: false,
+        momentarySwitchLongPress: false,
+        momentarySwitchMultiPress: false,
+    },
+
+    /** @see {@link MatterApplicationClusterSpecificationV1_0} § 1.11.5 */
+    attributes: {
+        /** Indicates the maximum number of positions the switch has with a minimum of 2 */
+        numberOfPositions: Attribute(0, TlvUInt8.bound({ min: 2 }), { default: 2 }),
```

Comment:
Please add an TODO "fixed" ... we change then when my PR is merged or I add there